### ISSUE TITLE
In.NET 5.0 WPF, a window in a container cannot call a style resource in app.xaml 

### ISSUE THREAD

**Issue by lifuhua123** [dotnet/wpf#4538]
In.NET 5.0 WPF, a window in a container cannot call a style resource in app.xaml ,
as shown in the figure :
![1](https://user-images.githubusercontent.com/33051107/118790163-16c65a00-b8c8-11eb-99e9-3a187dc03ceb.png)
![2](https://user-images.githubusercontent.com/33051107/118790174-18901d80-b8c8-11eb-92c0-389a1806287c.png)
![3](https://user-images.githubusercontent.com/33051107/118790181-1a59e100-b8c8-11eb-943e-8f5a596d7d8a.png)
![4](https://user-images.githubusercontent.com/33051107/118790193-1b8b0e00-b8c8-11eb-8ac0-993a53bd0ea8.png)

**Comment by miloush** [comment]
This reproduces in 4.8.

This seems to be a bug in the compiler actually. When `Application.StartupUri` is not in _App.xaml_, the compiler does not generate and include app.baml in the assembly resources.

**Comment by miloush** [comment]
...which seems to be a intentional optimization:

https://github.com/dotnet/wpf/blob/89d172db0b7a192de720c6cfba5e28a1e7d46123/src/Microsoft.DotNet.Wpf/src/PresentationBuildTasks/MS/Internal/MarkupCompiler/ParserExtension.cs#L163-L180

**Comment by miloush** [comment]
@lifuhua123 and this code also hints of a workaround: set any property on the `Application` element, for example explicitly include `ShutdownMode="OnLastWindowClose"`.

**Comment by ThomasGoulet73** [comment]
I think this issue is a duplicate of this one: https://github.com/dotnet/wpf/issues/4457. As I said in this comment: https://github.com/dotnet/wpf/issues/4457#issuecomment-827078499, the issue is caused by PresentationBuildTasks.dll so it will only apply to SDK-style csproj (.Net Framework or .Net Core) but this issue does not reproduce when using the old csproj style of .Net Framework.

**Comment by miloush** [comment]
This one reproduces for me with the old project file. That said, I agree this looks caused by the same underlying issue.

**Comment by ThomasGoulet73** [comment]
@miloush You're right, just tested it again and I was able to reproduce this issue and #4457 using the old project style. I think I just forgot to remove StartupUri in my test.

**Comment by miloush** [comment]
@ryalanms I think this is meant to be a bug report rather than a question, but perhaps can be closed as duplicate.

**Comment by lifuhua123** [comment]
@ThomasGoulet73 This problem is different from # 4457. Please look at the sample
[WpfApp2.zip](https://github.com/dotnet/wpf/files/6535666/WpfApp2.zip)

**Comment by lifuhua123** [comment]
@miloush You're right, I added shutdownMode =" onLastWindowClose "and the program starts, but why?

**Comment by ThomasGoulet73** [comment]
@lifuhua123 The problem is different but the cause is the same. You can also use the same workaround of adding another style to your resource dictionary in App.xaml. The bug is that it should generate the InitializeComponent method even if there is only one Style in the resource dictionary. By adding a property, you trigger the generation of InitializeComponent that was already suppose to occur when there is on style.

**Comment by lifuhua123** [comment]
Okay, so is this a problem or a bug?  I tried another method, the program can also start.  Set shutdownMode =" onLastWindowClose ", but don't know why?  @ThomasGoulet73
